src/library/scala/collection/mutable/PriorityQueue.scala

```diff
@@ -254,12 +320,24 @@ sealed class PriorityQueue[A](implicit val ord: Ordering[A])
    *
    *  @return  a priority queue with the same elements.
    */
-  override def clone(): PriorityQueue[A] = new PriorityQueue[A] ++= this.iterator
+  override def clone(): PriorityQueue[A] = {
+    val pq = new PriorityQueue[A]
+    for (i <- 1 to this.length) pq.unsafeAdd(resarr(i))
```

Comment:
This is way less bad than the previous version, but why not just copy the underlying array?  For instance, dynamically expanding the array rather than just setting it to the correct size results in twice the allocation needed and 2.4ish the number of copies (depending on the distribution of the lengths).